pythonwhat/State.py

```diff
@@ -44,13 +44,9 @@ def __init__(self, **kwargs):
 
         self.converters = None
 
-        self.pre_exercise_mappings = None
-        self.student_function_calls = None
-        self.solution_function_calls = None
-        self.student_mappings = None
-        self.solution_mappings = None
-        self.fun_usage = None
+        self.fun_usage = {}
         self.manual_sigs = None
+        self._parser_cache = {}
```

Comment:
the `_parser_cache` is cleaned up again as soon as you go into a substate right?